### DISCUSSION TITLE
How to avoid a blank render when using Mitsuba in Blender

### DISCUSSION THREAD

**Discussion by AnsonSavage** [mitsuba-renderer/mitsuba-blender#115] [Q&A] [answered]
Hello!

I successfully installed Mitsuba in Blender:

```Modules Installed (mitsuba-blender) from '/home/Downloads/Software/mitsuba-blender.zip' into '/home/.config/blender/4.2/scripts/addons'```

I then switch the render engine to Mitsuba to try and render the default cube, and I get the following error:
```
Traceback (most recent call last):
  File "/home/.config/blender/4.2/scripts/addons/mitsuba-blender/engine/final.py", line 42, in render
    self.converter.scene_to_dict(depsgraph)
TypeError: SceneConverter.scene_to_dict() missing 1 required positional argument: 'window_manager'
```

The rendering window appears, but the default cube does not (it only shows a transparent background). It seems like [this](https://github.com/mitsuba-renderer/mitsuba-blender/discussions/49#discussioncomment-5727754) user had a similar error.

Any thoughts?

Thanks!

**Comment by AnsonSavage** [comment]
Ah, it turns out that Mitsuba doesn't actually work as a render engine in Blender, even though it shoes up as a Render Engine. The main working functionality is the XML export.

See this answer: https://blender.stackexchange.com/questions/330254/mitsuba-3-for-blender

and this answer: https://github.com/mitsuba-renderer/mitsuba-blender/discussions/49

**Reply by merlinND** [reply, marked as answer]
Hello @AnsonSavage,

There is major work in https://github.com/mitsuba-renderer/mitsuba-blender/pull/121 to make Mitsuba a full-fledged render engine, but it hasn't been merged yet.
